Sources/SwiftFormat/PrettyPrint/TokenStreamCreator.swift

```diff
@@ -2460,39 +2460,85 @@ fileprivate final class TokenStreamCreator: SyntaxVisitor {
     return .visitChildren
   }
 
+  // Insert an `.escaped` break token after each series of whitespace in a substring
+  private func emitMultilineSegmentTextTokens(breakKind: BreakKind, segment: Substring) {
+    var currentWord = [Unicode.Scalar]()
+    var currentBreak = 0
+
+    func emitWord() {
+      if !currentWord.isEmpty {
+        var str = ""
+        str.unicodeScalars.append(contentsOf: currentWord)
+        appendToken(.syntax(str))
+        currentWord = []
+      }
+    }
+    func emitBreak() {
+      if currentBreak > 0 {
+        // We append this as a syntax, instead of a .space, so that it is always included in the output.
+        appendToken(.syntax(String(repeating: " ", count: currentBreak)))
+        appendToken(.break(breakKind, size: 0, newlines: .escaped))
+        currentBreak = 0
+      }
+    }
+
+    for scalar in segment.unicodeScalars {
+      // We don't have to worry about newlines occurring in segments.
+      // Either a segment will end in a newline character or the newline will be in trivia.
+      if scalar.properties.isWhitespace {
+        emitWord()
+        currentBreak += 1
+      } else {
+        emitBreak()
+        currentWord.append(scalar)
+      }
+    }
+
+    // Only one of these will actually do anything based on whether our last char was whitespace or not.
+    emitWord()
+    emitBreak()
+  }
+
   override func visit(_ node: StringSegmentSyntax) -> SyntaxVisitorContinueKind {
     // Looks up the correct break kind based on prior context.
+    func stringLiteralParent() -> StringLiteralExprSyntax? {
```

Comment:
This could just be a local variable rather than a nested function, I think?

Reply:
Looking back at the code, I don't remember why it was a nested function to begin with. It doesn't look like it depends on any state that would change elsewhere during execution.

I think it would be fine to simplify it.